packages/gatsby-plugin-sharp/src/scheduler.js

```diff
@@ -137,44 +130,41 @@ function runJobs(
   bar.total = totalJobs
 
   try {
-    const promises = processFile(
-      job.inputPath,
-      job.contentDigest,
-      jobs.map(job => job.job),
-      pluginOptions
-    ).map(promise =>
-      promise
-        .then(job => {
-          findDeferred(job).resolve()
-        })
-        .catch(err => {
-          findDeferred(job).reject({
-            err,
-            message: `Failed to process image ${job.inputPath}`,
-          })
+    worker({
+      inputPath: job.inputPath,
+      contentDigest: job.contentDigest,
+      transforms: jobs.map(child => {
+        return {
+          outputPath: child.job.outputPath,
+          transforms: child.job.args,
+        }
+      }),
+      pluginOptions,
+    })
+      .then(() => {
+        jobs.map(job => {
+          job.deferred.resolve()
         })
-        .then(() => {
-          imagesFinished += 1
 
-          // only show progress on build
-          if (reportStatus) {
-            bar.tick()
-          }
+        imagesFinished += jobs.length
 
-          boundActionCreators.setJob(
-            {
-              id: jobId,
-              imagesFinished,
-            },
-            { name: `gatsby-plugin-sharp` }
-          )
-        })
-    )
-
-    Promise.all(promises).then(() => {
-      boundActionCreators.endJob({ id: jobId }, { name: `gatsby-plugin-sharp` })
-      cb()
-    })
+        boundActionCreators.setJob(
+          {
+            id: jobId,
+            imagesFinished,
+          },
+          { name: `gatsby-plugin-sharp` }
+        )
+      })
+      .catch(() => {})
```

Comment:
This should probably reject with the previous message: `Failed to process image ${job.inputPath}`

This is also wrapped in a try catch block so perhaps we can do all error handling there?

Reply:
I guess you have to reject all job deferred here too

Reply:
I rewrote the pipeline :)